Functionality for plotting 2D data

wsp-mbuckley opened the pull request:
Describe Changes
---

- Added functions for creating subplots
- Added dataclasses for storing plot data
- Added functions for creating scatter, scatter density and hexbin plots
- Added matplotlib stylesheet for TfN (WIP!)

Task Checklist
----
- [ ] Have unittests been added (testing individual functions and their edge cases)
- [x] Have integration tests been added (if applicable, to test modules of functionality)
- [x] Updated RELEASE.md to reflect changes in PR
- [x] Have new dependencies been added?
  - [x] Have they been added to either `requirements.txt` or `requirements_dev.txt`.
  - [x] Have they been added to `setup.cfg`

BenTaylor-TfN commented on src/caf/viz/__init__.py:
```diff
@@ -1,3 +1,9 @@
 from ._version import __version__
+
+# Aliases
+from caf.viz.xy_plot import XYPlotType, axes_plot_xy, plot_xy
```
Does this need to be here? I feel like it sets a precedent of accessing these functions this way, when really they should be accessed through their module?

BenTaylor-TfN reviewed (commented):
I noticed there's a couple of TODOs in the code. Do any of these warrant being tracked as issues instead?

Is this basically ready to be merged into main and published?
I've set up the publishing workflow which means it should be automatically published to PyPI when we make a new release - just waiting on this code to test this out.

wsp-mbuckley commented:
@BenTaylor-TfN this should be ready for merge, we don't have proper unit tests yet but I've got a couple of integration tests. I've removed the TODO comments and added issues to the repo. Once merged I'll setup the release on GitHub and hopefully PyPi should pick that up
